numba/core/interpreter.py

```diff
@@ -79,6 +79,432 @@ def get_assignment_source(self, destname):
+                    ):
+                        not_found = False
+                    else:
+                        varkwarg_loc -= 1
+                if (
+                    kws
+                    or not_found
+                    or not (
+                        isinstance(keyword_def.value, ir.Expr)
+                        and keyword_def.value.op == "build_map"
+                    )
+                ):
+                    # If we couldn't find where the kwargs are created
+                    # then it should be a normal **kwargs call
+                    # so we produce an unsupported message.
+                    raise UnsupportedError(errmsg)
+                # Determine the kws
+                if keyword_def.value.items:
+                    # n_kws <= 15 case.
+                    kws = _call_function_ex_replace_kws_small(
+                        keyword_def.value,
+                        new_body,
+                        varkwarg_loc,
+                    )
+                else:
+                    # n_kws > 15 case.
+                    kws = _call_function_ex_replace_kws_large(
+                        blk.body,
+                        varkwarg.name,
+                        varkwarg_loc,
+                        i - 1,
+                        new_body,
+                    )
+                start_search = varkwarg_loc
+                # Vararg isn't required to be provided.
+                if vararg is not None:
+                    if args:
+                        # If we have vararg then args is expected to
+                        # be an empty list.
+                        raise UnsupportedError(errmsg)
+                    vararg_loc = start_search
+                    args_def = None
+                    not_found = True
+                    while vararg_loc >= 0 and not_found:
+                        args_def = blk.body[vararg_loc]
+                        if (
+                            isinstance(args_def, ir.Assign)
+                            and args_def.target.name == vararg.name
+                        ):
+                            not_found = False
+                        else:
+                            vararg_loc -= 1
+                    if not_found:
+                        # If we couldn't find where the args are created
+                        # then we can't handle this format.
+                        raise UnsupportedError(errmsg)
+                    if (
+                        isinstance(args_def.value, ir.Expr)
+                        and args_def.value.op == "build_tuple"
+                    ):
+                        # n_args <= 30 case.
+                        args = _call_function_ex_replace_args_small(
+                            args_def.value,
+                            new_body,
+                            vararg_loc
+                        )
+                    else:
+                        args = _call_function_ex_replace_args_large(
```

Comment:
Add comment on the expected IR node for args_def.

Reply:
I added a comment explaining my assumptions. Please let me know if it needs further clarification.